### PR TITLE
Bring multiarch support to the build-sync job

### PR DESCRIPTION
**Pull request by tbielawa**
Supporting work for https://jira.coreos.com/browse/ART-837 "configure
per-arch processes for creating a nightly release image"

### CODE REVIEW

File: pipeline-scripts/buildlib.groovy [hunk 1]
```diff
@@ -812,6 +812,24 @@ def parse_record_log( working_dir ) {
...
 
+// Find the supported arches for this release
+//
+// @param branch <String>: The name of the branch to get configs
+//   for. For example: 'openshift-4.3'
+//
+// @return arches <List>: A list of the arches built for this branch
+def branch_arches(String branch) {
+    echo("Fetching group config for '${branch}'")
+    def groupConfigSource = "https://gitlab.cee.redhat.com/openshift-art/ocp-build-data/raw/${branch}/group.yml"
```
Comment (tbielawa) [inline pipeline-scripts/buildlib.groovy]: This needs to change back to github, doesn't it?

Comment (sosiouxme) [reply pipeline-scripts/buildlib.groovy]: yep https://github.com/openshift/aos-cd-jobs/pull/1997

File: jobs/build/build-sync/Jenkinsfile [hunk 2]
```diff
@@ -18,105 +19,80 @@ node {
             [
                 $class : 'ParametersDefinitionProperty',
                 parameterDefinitions: [
+                    commonlib.suppressEmailParam(),
```
Comment (sosiouxme) [inline jobs/build/build-sync/Jenkinsfile]: :+1:

File: jobs/build/build-sync/Jenkinsfile [hunk 3]
```diff
@@ -18,105 +19,80 @@ node {
...
-    def baseImageStream = "/home/jenkins/base-art-latest-imagestream-${params.BUILD_VERSION}.yaml"
-    // Kubeconfig allowing ART to interact with api.ci.openshift.org
-    def ciKubeconfig = "/home/jenkins/kubeconfigs/art-publish.kubeconfig"
-
-    // See 'oc image mirror --help' for more information.
-    // This is the template for the SRC=DEST strings mentioned above.
-    def ocFmtStr = "registry.reg-aws.openshift.com:443/{repository}=quay.io/openshift-release-dev/ocp-v4.0-art-dev:{version}-{release}-{image_name_short}"
-
-    buildlib.cleanWorkdir(mirrorWorking)
+    commonlib.checkMock()
```
Comment (sosiouxme) [inline jobs/build/build-sync/Jenkinsfile]: :+1: good grief

File: pipeline-scripts/buildlib.groovy (hunk 1 above)
Comment (tbielawa) [reply pipeline-scripts/buildlib.groovy]: Merged that pr. and fixed this

**Pull request merged**